source/server/configuration_impl.cc

```diff
@@ -51,6 +52,12 @@ void MainImpl::initialize(const envoy::config::bootstrap::v2::Bootstrap& bootstr
       server.localInfo(), server.accessLogManager(), server.admin());
   const auto& listeners = bootstrap.static_resources().listeners();
   ENVOY_LOG(info, "loading {} listener(s)", listeners.size());
+
+  // Immediately pause RDS to ensure that we don't send any requests until we've
+  // subscribed to all the RDS resources. The subscriptions happen in the init callbacks,
+  // so we pause RDS until we've completed all the callbacks.
+  cluster_manager_->adsMux().pause(Config::TypeUrl::get().RouteConfiguration);
```

Comment:
I don't have a great answer, but I wish there was a more general way of solving this problem, as I could see us having issues in the future for other resource types that get added. @htuch any ideas?

Reply:
I think it's fine, there are only two other places this comes up: EDS and SDS. We should either fix EDS in this PR or leave some TODOs and an open issue. I don't think we're going to arbitrarily scale the number of resources in xDS, so seems fine to have a handful of special case handling.